### PR TITLE
[HAL-1070] Web Console use select boxes for inputs which must be a particular value - tests

### PR DESCRIPTION
**Pull request by honza-kasik**
Added test for a new type-ahead feature

### CODE REVIEW

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/suggestbox/SuggestBoxPopUp.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        return elements.stream()
+                .map(SuggestBoxPopUpValue::new)
+                .collect(Collectors.toList());
+    }
+
+    public List<String> getLabelValues() {
+        return getValues().stream()
+                .map(SuggestBoxPopUpValue::toString)
+                .collect(Collectors.toList());
+    }
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/suggestbox/SuggestBoxPopUp.java:45]: Please provide JavaDoc for all public methods

File: common/src/main/java/org/jboss/hal/testsuite/util/Console.java [hunk 2]
```diff
@@ -77,30 +78,57 @@ public void waitUntilFinished() {
...
+    /**
+     * Retrieves currently opened popup menu which class is {@link PopUpFragment.ROOT_SELECTOR}, root
+     * {@link PopUpFragment.ROOT_SELECTOR} and class {@link PopUpFragment.class}.
+     * @return
+     */
     public PopUpFragment openedPopup() {
         return openedPopup(PopUpFragment.class, PopUpFragment.ROOT_SELECTOR);
     }
 
+    public SuggestBoxPopUp openedSuggestionBoxPopUp() {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/Console.java:131]: Please provide JavaDoc

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionChecker.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ *         Created on 8/25/16.
+ */
+public class SuggestionChecker {
+
+    private WebDriver browser;
+    private ConfigFragment configFragment;
+    private String suggestBoxInputLabel;
+    private SuggestionResource suggestionResource;
+
+    public SuggestionChecker(Builder builder) {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionChecker.java]: Make it private

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    private OnlineManagementClient client;
+    private List<Address> templates;
+
+    private SuggestionResource(Builder builder) {
+        this.client = builder.client;
+        this.templates = builder.templates;
+    }
+
+    public List<String> readSuggestions() throws IOException {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java:40]: JavaDoc please

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            if (this.templates == null) {
+                this.templates = new LinkedList<>();
+            }
+            this.templates.add(template);
+            return this;
+        }
+
+        public SuggestionResource build() {
+            return new SuggestionResource(this);
+        }
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java:119]: Check whether all mandatory parameters are defined

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+/**
+ * Class providing suggestions for given resources identified by templates
+ *
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 8/22/16.
+ */
+public class SuggestionResource {
+
+    private OnlineManagementClient client;
+    private List<Address> templates;
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java]: Make them final

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        private OnlineManagementClient client;
+        private List<Address> templates;
+
+        public Builder() {
+        }
+
+        public Builder withClient(OnlineManagementClient client) {
+            this.client = client;
+            return this;
+        }
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java:91]: JavaDoc please

**Comment by rhatlapa** [comment]
Generally LGTM, It would be nice to improve some JavaDoc. 

The test coverage could be also increased, still that is not blocker for this PR.
Here are some of my suggestions for improving test coverage:
1) Provide negative test cases => strings, for which there shouldn't be any suggestions or even provide strings with stranger characters, such as with `*` and observe what will happen.
2) Also there could be added case, where you have already typed string, which suggests you some options, then you add letter and checking whether the suggestions are properly updated. The same with removing single letter.

**Comment by honza-kasik** [comment]
Added Javadoc to all public methods and worked in suggested scenarios (at least smokes). Some test are currently failing due to the [HAL-1169](https://issues.jboss.org/browse/HAL-1169).

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionChecker.java [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+        Editor editor = configFragment.getEditor();
+        if (!configFragment.isInEditMode()) {
+            editor = configFragment.edit();
+        }
+        return editor;
+    }
+
+    public class SuggestState {
+
+        private String inputFieldValue = "";
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionChecker.java]: No need to initialize to empty String as it is always set in constructor. Also make it final.

File: common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.jboss.as.cli.Util.ADDRESS;
+import static org.jboss.as.cli.Util.RESULT;
+
+/**
+ * Class providing suggestions for given resources identified by templates
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/util/SuggestionResource.java]: You should explain what template is. That it is actually an `Address`

**Comment by rhatlapa** [comment]
Just minor comments, please update and I shall merge.

**Comment by honza-kasik** [comment]
Suggestion worked in, thanks!

File: common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/suggestbox/SuggestedItemNotFoundException.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package org.jboss.hal.testsuite.fragment.shared.modal.suggestbox;
+
+import org.openqa.selenium.NotFoundException;
+
+/**
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 8/22/16.
+ */
+public class SuggestedItemNotFoundException extends NotFoundException {
```
Comment (rhatlapa) [inline common/src/main/java/org/jboss/hal/testsuite/fragment/shared/modal/suggestbox/SuggestedItemNotFoundException.java:10]: The Javadoc above should be updated to reflect the class => exception usage

**Comment by rhatlapa** [comment]
I've found one small thing, which I missed previously. Please update and squash the fixing commits and I shall merge.

**Comment by honza-kasik** [comment]
Fixed and squashed, thanks!

**Comment by rhatlapa** [comment]
Thanks, LGTM, merging.

**Pull request merged**